src/scheduler/tasks/SynchronizeTagsImportTask.ts

```diff
@@ -0,0 +1,75 @@
+import Constants from '../../utils/Constants';
+import { ImportStatus } from '../../types/GlobalType';
+import { LockEntity } from '../../types/Locking';
+import LockingManager from '../../locking/LockingManager';
+import Logging from '../../utils/Logging';
+import SchedulerTask from '../SchedulerTask';
+import { ServerAction } from '../../types/Server';
+import Tag from '../../types/Tag';
+import TagStorage from '../../storage/mongodb/TagStorage';
+import TagValidator from '../../server/rest/v1/validator/TagValidation';
+import { TaskConfig } from '../../types/TaskConfig';
+import Tenant from '../../types/Tenant';
+
+const MODULE_NAME = 'SynchronizeUsersImportTask';
+
+export default class SynchronizeTagsImportTask extends SchedulerTask {
+  async processTenant(tenant: Tenant, config: TaskConfig): Promise<void> {
+    const synchronizeTagsImport = LockingManager.createExclusiveLock(tenant.id, LockEntity.TAGS, 'synchronize-tags-import');
+    if (await LockingManager.acquire(synchronizeTagsImport)) {
+      try {
+        const importedTags = await TagStorage.getImportedTags(tenant.id, { statuses: [ImportStatus.UNKNOWN] }, Constants.DB_PARAMS_MAX_LIMIT);
+        if (importedTags.count !== 0) {
+          for (const importedTag of importedTags.result) {
+            const foundTag = await TagStorage.getTag(tenant.id, importedTag.id);
+            if (foundTag) {
+              importedTag.status = ImportStatus.ERROR;
+              importedTag.error = 'Tag id already exists';
+              await TagStorage.saveImportedTag(tenant.id, importedTag);
+            } else {
+              const tag: Tag = {
+                id: importedTag.id,
+                description: importedTag.description,
+                issuer: true,
+                deleted: false,
+                active: true,
+                createdBy: { id: importedTag.importedBy },
```

Comment:
I think you can assign right away the ID as a string:
`createdBy: importedTag.importedBy,`